homeassistant/components/volvo/sensor.py

```diff
@@ -0,0 +1,360 @@
+"""Volvo sensors."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from volvocarsapi.models import (
+    VolvoCarsApiBaseModel,
+    VolvoCarsValue,
+    VolvoCarsValueField,
+    VolvoCarsVehicle,
+)
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    PERCENTAGE,
+    EntityCategory,
+    UnitOfElectricCurrent,
+    UnitOfEnergy,
+    UnitOfEnergyDistance,
+    UnitOfLength,
+    UnitOfSpeed,
+    UnitOfTime,
+    UnitOfVolume,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .const import DATA_BATTERY_CAPACITY
+from .coordinator import VolvoConfigEntry
+from .entity import VolvoEntity, VolvoEntityDescription, value_to_translation_key
+
+PARALLEL_UPDATES = 0
+
+# Entities having an "unknown" value should report None as the state
+_UNKNOWN_VALUES = [
+    "UNSPECIFIED",
+    "CONNECTION_STATUS_UNSPECIFIED",
+    "CHARGING_SYSTEM_UNSPECIFIED",
+]
+
+
+@dataclass(frozen=True, kw_only=True)
+class VolvoSensorDescription(VolvoEntityDescription, SensorEntityDescription):
+    """Describes a Volvo sensor entity."""
+
+    value_fn: Callable[[VolvoCarsValue], Any] | None = None
+    available_fn: Callable[[VolvoCarsVehicle], bool] = lambda vehicle: True
+
+
+def _availability_status(field: VolvoCarsValue) -> str:
+    reason = field.get("unavailable_reason")
+    return reason if reason else str(field.value)
+
+
+def _calculate_time_to_service(field: VolvoCarsValue) -> int:
+    value = int(field.value)
+
+    # Always express value in days
+    if isinstance(field, VolvoCarsValueField) and field.unit == "months":
+        return value * 30
+
+    return value
+
+
+_DESCRIPTIONS: tuple[VolvoSensorDescription, ...] = (
+    # command-accessibility endpoint
+    VolvoSensorDescription(
+        key="availability",
+        translation_key="availability",
+        api_field="availabilityStatus",
+        device_class=SensorDeviceClass.ENUM,
+        options=[
+            "available",
+            "car_in_use",
+            "no_internet",
+            "power_saving_mode",
+            "unspecified",
```

Comment:
I just saw that list where you mention that we map this unknown, is that also applicable here?